integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/testsupport/util/ElasticsearchTckBackendFeatures.java

```diff
@@ -200,6 +201,19 @@ public boolean supportsHighlighterPlainOrderByScoreMultivaluedField() {
 		);
 	}
 
+	@Override
+	public boolean supportsHighlighterUnifiedPhraseMatching() {
+		return isActualVersion(
+				esVersion -> !esVersion.isLessThan( "8.10" ) || isLatestSnapshotContainer(),
+				osVersion -> false
+		);
+	}
+
+	public static boolean isLatestSnapshotContainer() {
+		return System.getProperty( "org.hibernate.search.integrationtest.backend.elasticsearch.image.tag", "" )
+				.toLowerCase( Locale.ROOT ).contains( "snapshot" );
+	}
+
```

Comment:
This should probably be in `ElasticsearchTestDialect`?

And TBH in this case I'm wondering if we shouldn't rather try to guess the version in `ElasticsearchTestDialect` by sending a request to the container in `getActualVersion`... I.e.:


Unset the version in Maven properties:

```
-Dtest.elasticsearch.run.version=
```

Add this in `ElasticsearchTestDialect#getActualVersion`:

```
if ( distribution != AMAZON_OPENSEARCH_SERVERLESS && (version == null || version.isBlank()) ) {
    try ( TestElasticsearchClient client = new TestElasticsearchClient() ) {
        client.open(...); 
        // This will probably use org.hibernate.search.backend.elasticsearch.client.impl.ElasticsearchClientUtils#tryGetElasticsearchVersion
        version = client.callSomeMethodToGetTheVersion();
    }
}
```

Reply:
mmmm, ok I'll give that a try 😃